File: ios/MoodleLMS/Core/Network/DTO/SectionDTO.swift
import Foundation

struct SectionDTO: Decodable {
    let id: Int
    let name: String
    let summary: String?
    let visible: Int?
    let modules: [ModuleDTO]?
}

struct ModuleDTO: Decodable {
    let id: Int
    let name: String
    let url: String?
    let modname: String?
    let modplural: String?
    let modicon: String?
    let visible: Int?
}

File: android/app/src/main/java/io/itcorner/moodle/core/network/dto/SectionDto.kt
package io.itcorner.moodle.core.network.dto

import kotlinx.serialization.Serializable

@Serializable
data class SectionDto(
    val id: Long,
    val name: String,
    val summary: String? = null,
    val visible: Int? = null,
    val modules: List<ModuleDto> = emptyList(),
)

@Serializable
data class ModuleDto(
    val id: Long,
    val name: String,
    val url: String? = null,
    val modname: String? = null,
    val modplural: String? = null,
    val modicon: String? = null,
    val visible: Int? = null,
)
